src/lib/storage/dictionary_segment.hpp

```diff
@@ -86,7 +95,8 @@ class DictionarySegment : public BaseDictionarySegment {
 
  protected:
   const std::shared_ptr<const pmr_vector<T>> _dictionary;
-  const std::shared_ptr<const BaseCompressedVector> _attribute_vector;
+  std::shared_ptr<const std::span<const T>> _dictionary_span;
+  std::shared_ptr<const BaseCompressedVector> _attribute_vector;
```

Comment:
Eventually the span should be named _dictionary. The corresponding data member maybe "dictionary_data"?

Reply:
Good idea!